tests/Unit/Evolution/Systems/ScalarWave/Test_Equations.cpp

```diff
@@ -249,5 +250,95 @@ SPECTRE_TEST_CASE("Unit.Evolution.Systems.ScalarWave.UpwindFlux",
+  MAKE_GENERATOR(generator);
+  std::uniform_real_distribution<> dist(0.1, 1.);
+
+  const auto unit_normal_int =
+      make_with_random_values<tnsr::i<DataVector, Dim, Frame::Inertial>>(
+          make_not_null(&generator), make_not_null(&dist), used_for_size);
+  const auto unit_normal_ext = [&unit_normal_int, &npts]() {
+    tnsr::i<DataVector, Dim> normal{pow<Dim>(npts)};
+    for (size_t i = 0; i < Dim; ++i) {
+      normal.get(i) = -unit_normal_int.get(i);
+    }
+    return normal;
+  }();
+
+  const auto n_dot_flux_pi_int = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+  const auto n_dot_flux_phi_int =
+      make_with_random_values<tnsr::i<DataVector, Dim, Frame::Inertial>>(
+          make_not_null(&generator), make_not_null(&dist), used_for_size);
+  const auto v_plus_int = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+  const auto v_minus_int = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+
+  const auto n_dot_flux_pi_ext = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+  const auto n_dot_flux_phi_ext =
+      make_with_random_values<tnsr::i<DataVector, Dim, Frame::Inertial>>(
+          make_not_null(&generator), make_not_null(&dist), used_for_size);
+  const auto v_plus_ext = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+  const auto v_minus_ext = make_with_random_values<Scalar<DataVector>>(
+      make_not_null(&generator), make_not_null(&dist), used_for_size);
+
+  // package data needed for flux
+  ScalarWave::PenaltyFlux<Dim> flux_computer{};
+  flux_computer.package_data(make_not_null(&packaged_data_int),
+                             n_dot_flux_pi_int, n_dot_flux_phi_int, v_plus_int,
+                             v_minus_int, unit_normal_int);
+  flux_computer.package_data(make_not_null(&packaged_data_ext),
+                             n_dot_flux_pi_ext, n_dot_flux_phi_ext, v_plus_ext,
+                             v_minus_ext, unit_normal_ext);
+
+  Scalar<DataVector> normal_dot_numerical_flux_pi(pow<Dim>(npts), 0.0);
+  Scalar<DataVector> normal_dot_numerical_flux_psi(pow<Dim>(npts), 0.0);
+  tnsr::i<DataVector, Dim, Frame::Inertial> normal_dot_numerical_flux_phi(
+      pow<Dim>(npts), 0.0);
+  apply_numerical_flux(flux_computer, packaged_data_int, packaged_data_ext,
+                       make_not_null(&normal_dot_numerical_flux_pi),
+                       make_not_null(&normal_dot_numerical_flux_phi),
+                       make_not_null(&normal_dot_numerical_flux_psi));
+
+  CHECK(normal_dot_numerical_flux_psi ==
+        Scalar<DataVector>(pow<Dim>(npts), 0.0));
+  CHECK(normal_dot_numerical_flux_pi ==
```

Comment:
Can you do these checks with python? See, e.g. `Test_Hll.cpp`

Reply:
I think this is the only thing left, everything else LGTM :)

Reply:
Changed to using `pypp`